docs/index.rst

```diff
@@ -3,185 +3,288 @@ The confluent_kafka API
 
 A reliable, performant and feature rich Python client for Apache Kafka v0.8 and above.
 
-Clients
+Client API
+   - :ref:`AdminClient <pythonclient_adminclient>`
    - :ref:`Consumer <pythonclient_consumer>`
+   - :ref:`DeserializingConsumer <serde_consumer>`
    - :ref:`Producer <pythonclient_producer>`
-   - :ref:`AdminClient <pythonclient_adminclient>`
+   - :ref:`SerializingConsumer <serde_producer>`
+   - :ref:`SchemaRegistry <schemaregistry_client>`
+
+   - :ref:`AvroConsumer(legacy) <avro_consumer>`
+   - :ref:`AvroProducer(legacy) <avro_producer>`
+
+Serialization API
+    - :ref:`Deserializer <serde_deserializer>`
+    - :ref:`AvroDeserializer <schemaregistry_avro_deserializer>`
+    - :ref:`JSONDeserializer <schemaregistry_json_deserializer>`
+    - :ref:`ProtobufDeserializer <schemaregistry_protobuf_deserializer>`
+
+    - :ref:`Serializer <serde_Serializer>`
+    - :ref:`AvroSerializer <schemaregistry_avro_serializer>`
+    - :ref:`JSONSerializer <schemaregistry_json_serializer>`
+    - :ref:`ProtobufSerializer <schemaregistry_protobuf_serializer>`
 
+:ref:`Transactional Producer API <pythonclient_transactional>`
+
+:ref:`Kafka Client Configuration <pythonclient_configuration>`
 
 Supporting classes
     - :ref:`Message <pythonclient_message>`
     - :ref:`TopicPartition <pythonclient_topicpartition>`
+    - :ref:`SerializationContext <serde_ctx>`
```

Comment:
The schemas are not really used by the application, so dont think we need to include them in the TOC here.

Reply:
If you extend the serializer it might which is why it's in supporting classes